Parsers/ASimWebSession/Parsers/ASimWebSessionBarracudaWAF.yaml

```diff
@@ -0,0 +1,305 @@
+          User = SrcUsername,
+          IpAddr = SrcIpAddr,
+          Src = SrcIpAddr,
+          HttpStatusCode = EventResultDetails
+      | project-away
+          *_d,
+          *_s,
+          _ResourceId,
+          severity,
+          status_code,
+          RawData,
+          EventResult_*,
+          SourceIP,
+          Message,
+          EventType_lookup,
+          Computer,
+          MG,
+          ManagementGroupName,
+          TenantId,
+          SourceSystem
+      )
+  };
+  let BarracudaCEF = (disabled: bool=false) {
+      CommonSecurityLog
+      | where not(disabled) and DeviceVendor startswith "Barracuda" and (DeviceProduct == "WAF" or DeviceProduct == "WAAS")
+      | where DeviceEventCategory in ("WF", "TR")
+      | lookup EventResultWFLookup on $left.DeviceAction == $right.Action_s
+      | lookup EventTypeLookup on $left.DeviceEventCategory == $right.LogType_s
+      | extend
+          EventType = EventType_lookup,
+          severity = toint(LogSeverity)
+      | lookup SeverityLookup on severity
+      | extend
+          Dst = DestinationIP,
+          EventCount = toint(1),
+          EventProduct = "WAF",
+          EventSchema = "WebSession",
+          EventSchemaVersion = "0.2.6",
+          EventVendor = "Barracuda",
+          status_code = toint(EventOutcome)
+      | extend
+          EventResult_TR = case(
+                      status_code between (200 .. 299),
+                      "Success", 
+                      status_code between (400 .. 599),
+                      "Failure",
+                      status_code between (300 .. 399),
+                      "Partial",
+                      "NA"
+                  ),
+          RuleName = iff(DeviceEventCategory == "WF", DeviceCustomString3, "")
+      | extend
+          Dvc = DeviceName,
+          EventResult = iff(DeviceEventCategory == "TR", EventResult_TR, EventResult_WF),
+          EventStartTime = iff(isnotempty(FlexNumber2), unixtime_milliseconds_todatetime(tolong(ReceiptTime)-tolong(FlexNumber2)), unixtime_milliseconds_todatetime(tolong(ReceiptTime))),
+          DstIpAddr = DestinationIP,
+          SrcIpAddr = SourceIP,
+          DstBytes = tolong(ReceivedBytes),
+          DstPortNumber = toint(coalesce(DestinationPort,FieldDeviceCustomNumber1)),
+          HttpCookie = RequestCookies,
+          HttpReferrer = RequestContext,
+          HttpRequestBodyBytes = tolong(ReceivedBytes),
+          HttpRequestMethod = RequestMethod,
+          HttpResponseBodyBytes = tolong(SentBytes),
+          NetworkDuration = toint(FlexNumber2),
+          HttpUserAgent = RequestClientApplication,
+          NetworkSessionId = SourceUserID,
+          Rule = RuleName,
+          SrcPortNumber = toint(SourcePort),
+          SrcUsername = SourceUserName,
+          DstUsername = DestinationUserName,
+          Url = RequestURL,
+          HttpResponseCacheControl = iff(
+                                DeviceCustomNumber2 == 0,
```

Comment:
'FieldDeviceCustomNumber' for this not needed?

Reply:
Done